### PR TITLE
#71 - annual summary

### LINKED ISSUE #71: Holidays summary calendar view
It would be better if under `/holidays` employee will be able to see calendar for the whole year with marked free days. It may be non-interactive calendar.

Something like:
image

### LINKED ISSUE #82: More details in vacation calendar
As an employee I want to click on the icon in vacation calendar and see details of vacation request. I should only see details of my vacations. Admins, technical/administrative approvers should see everyone's vacation details.

### PR DESCRIPTION
**Pull request by Baakoma**
Me and @EwelinaLasowy created a brand new summary page for employees. On this page, employees can see their vacations (approved and pending) and holidays with additional information by hovering over them.
image

It should close #71 and should close #82

### CODE REVIEW

File: app/Domain/Validation/Rules/DoesNotExceedLimitRule.php [hunk 1]
```diff
@@ -41,13 +41,16 @@ public function errorMessage(): string
 
     protected function getUserVacationLimit(User $user, YearPeriod $yearPeriod): int
     {
-        return $user->vacationLimits()->where("year_period_id", $yearPeriod->id)->first()->days ?? 0;
+        return $user->vacationLimits()
+            ->whereBelongsTo($yearPeriod)
+            ->first()
+            ->days ?? 0;
```
Comment (mtracz) [inline app/Domain/Validation/Rules/DoesNotExceedLimitRule.php:47]: I think `?` is missing
```?->days ?? 0```
Cause `first()` method may return null, then you will receive something like "call property on null"

File: app/Eloquent/Models/Vacation.php [hunk 2]
```diff
@@ -50,4 +50,12 @@ public function scopeApproved(Builder $query): Builder
             fn(Builder $query) => $query->states(VacationRequestStatesRetriever::successStates()),
         );
     }
+
+    public function scopePending(Builder $query): Builder
+    {
+        return $query->whereRelation(
+            "vacationRequest",
+            fn(Builder $query) => $query->states(VacationRequestStatesRetriever::pendingStates()),
```
Comment (mtracz) [inline app/Eloquent/Models/Vacation.php:58]: Could you add return type of arrow function?

File: app/Infrastructure/Http/Controllers/AnnualSummaryController.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            ->vacations()
+            ->with("vacationRequest.vacations")
+            ->whereBelongsTo($yearPeriod)
+            ->pending()
+            ->get();
+
+        return inertia("AnnualSummary", [
+            "holidays" => $holidays->mapWithKeys(fn(Holiday $holiday) => [$holiday->date->toDateString() => $holiday->name]),
+            "vacations" => $vacations->mapWithKeys(fn(Vacation $vacation) => [$vacation->date->toDateString() => new SimpleVacationRequestResource($vacation->vacationRequest)]),
+            "pendingVacations" => $pendingVacations->mapWithKeys(fn(Vacation $vacation) => [$vacation->date->toDateString() => new SimpleVacationRequestResource($vacation->vacationRequest)]),
```
Comment (mtracz) [inline app/Infrastructure/Http/Controllers/AnnualSummaryController.php:40]: Could you add return types for arrow functions?

**Review by mtracz — CHANGES_REQUESTED** [review]

**Comment by mtracz** [comment]
What will happen with `/holidays` route / page?

**Comment by Baakoma** [comment]
> What will happen with `/holidays` route / page?

I think `/holidays` will be still available

**Review by mtracz — APPROVED** [review]

**Pull request merged**